src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/TransformFunctions.kt

```diff
@@ -157,9 +158,20 @@ fun transform(licence: EntityLicence): ModelLicence {
     isVariation = licence.variationOfId != null,
     variationOf = licence.variationOfId,
     createdByFullName = "${licence.createdBy?.firstName} ${licence.createdBy?.lastName}",
+    isInPssPeriod = checkLicenceIsInPssPeriod(licence),
   )
 }
 
+fun checkLicenceIsInPssPeriod(licence: EntityLicence): Boolean {
+  val led = licence.licenceExpiryDate
+  val tused = licence.topupSupervisionExpiryDate
+
+  if (led != null && tused != null) {
+    return led.isBefore(LocalDate.now()) && !(tused.isBefore(LocalDate.now()))
+  }
+  return false
+}
```

Comment:
maybe this could live on the EntityLicence as an `isInPssPeriod` function? - I find it's generally good to push code down layers as it tends to become easier to test. 

Then here you would just need to copy across the field.

this would be nice as it means we have business logic closer to the domain entity.

Reply:
moved isInPssPeriod to EntityLicence.